app/policies/roled_controller_policy.rb

```diff
@@ -0,0 +1,54 @@
+class RoledControllerPolicy
+  attr_reader :api_user, :resource_class, :resource_name, :action_name, :params, :scope_context, :add_active_resources_scope
+  
+  def initialize(api_user, resource_class, resource_name, action_name, params, scope_context: {}, add_active_resources_scope: true, **kwargs)
+    @api_user = api_user
+    @resource_class = resource_class
+    @resource_name = resource_name
+    @action_name = action_name
+    @params = params
+    @scope_context = scope_context
+    @add_active_resources_scope = add_active_resources_scope
+  end
+
+  def resources_exist?
+    resource_ids.blank? ? true : scope.exists?
+  end
+
+  def scope
+    @scope ||= api_user.scope(klass: resource_class,
+                              action: controlled_scope,
+                              ids: resource_ids,
+                              context: scope_context,
+                              add_active_scope: add_active_resources_scope)
+  end
+
+  def resource_ids
+    @resource_ids ||= array_id_params(_resource_ids)
+  end
+
+  private
+
+  def controlled_scope
```

Comment:
minor - Should we rename this to `controlled_action` while you are at it?

Reply:
This method is refactored away in `pundit-scopes` follow-up branch.

Reply:
This method is refactored away in `pundit-scopes` follow-up branch.